rust/datafusion/src/execution/context.rs

```diff
@@ -289,14 +289,12 @@ impl ExecutionContext {
 
     /// Deregisters the named table.
     ///
-    /// Returns true if the table was successfully de-reregistered.
-    pub fn deregister_table(&mut self, name: &str) -> bool {
-        self.state
-            .lock()
-            .unwrap()
-            .datasources
-            .remove(&name.to_string())
-            .is_some()
+    /// Returns the registered provider, if any
+    pub fn deregister_table(
+        &mut self,
+        name: &str,
+    ) -> Option<Arc<dyn TableProvider + Send + Sync>> {
```

Comment:
This changes the signature of `deregister_table` to be consistent with other Rust ecosystem APIs and returned the previously registered item, if any